frontend/src/composables/use-sensitive-media.ts

```diff
@@ -0,0 +1,74 @@
+import { computed } from "vue"
+
+import type { Media } from "~/types/media"
+
+import { useUiStore } from "~/stores/ui"
+
+type SensitiveVisibilityState =
+  | "insensitive"
+  | "sensitive-shown"
+  | "sensitive-hidden"
+
+/**
+ * A helper composable for working with sensitive media.
+ * Provides the current visibility of a sensitive media,
+ * along with toggles for hiding/revealing the media.
+ */
+export function useSensitiveMedia(
+  media: Pick<Media, "id" | "isSensitive"> | null
+) {
+  const uiStore = useUiStore()
+
+  /**
+   * The current state of a single sensitive media item.
+   * Respects defaults from the site's filters and is
+   * updated by user interactions.
+   */
+  const visibility = computed<SensitiveVisibilityState>(() => {
+    if (!media) {
+      return "insensitive"
```

Comment:
Nit: maybe "non-sensitive" is a better name for this?

Reply:
Agreed. Updated in c8e6917b2